src/neon/stage1.rs

```diff
@@ -113,14 +113,9 @@ impl Default for Utf8CheckingState {
 #[inline]
 fn is_utf8_status_ok(has_error: int8x16_t) -> bool {
     unsafe {
-        let utf8_error_bits: u128 = mem::transmute(
-            vandq_s16(
-                mem::transmute(has_error),
-                mem::transmute(has_error)
-            )
-        );
+        let has_error_128 : i128 = mem::transmute(has_error);
 
-        utf8_error_bits as u16 == 0
+        has_error_128 as u32 == 0
```

Comment:
I don't think any type cast should happen here this line should be
```rust
has_error_128  == 0
```
as has_error is a `int8x16_t` so 16*8 bit or 128 bit in total. downcasting that to anything else will ignore all the upper bits.

Reply:
success!